### PR TITLE
Make the state machine naming for MTRDevice match reality better.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Basically covers my post-landing review comments on https://github.com/project-chip/connectedhomeip/pull/32579

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRDevice.mm [hunk 1]
```diff
@@ -1025,20 +1051,19 @@ - (void)_setupSubscription
...
     MTRWeakReference<MTRDevice *> * weakSelf = [MTRWeakReference weakReferenceWithObject:self];
-    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t) (kTimeToWaitBeforeMarkingUnreachableAfterSettingUpSubscription * NSEC_PER_SEC)), self.queue, ^{
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, static_cast<int64_t>(kSecondsToWaitBeforeMarkingUnreachableAfterSettingUpSubscription) * static_cast<int64_t>(NSEC_PER_SEC)), self.queue, ^{
         MTRDevice * strongSelf = weakSelf.strongObject;
         if (strongSelf != nil) {
-            os_unfair_lock_lock(&strongSelf->_lock);
-            [strongSelf _markDeviceAsUnreachableIfNotSusbcribed];
-            os_unfair_lock_unlock(&strongSelf->_lock);
+            std::lock_guard(strongSelf->_lock);
+            [strongSelf _markDeviceAsUnreachableIfNeverSubscribed];
```
Comment (woody-apple) [inline src/darwin/Framework/CHIP/MTRDevice.mm:1066]: Does this compile? I was getting compilation errors with this

**Review by woody-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRDevice.mm [hunk 2]
```diff
@@ -1025,20 +1051,19 @@ - (void)_setupSubscription
...
     // Set up a timer to mark as not reachable if it takes too long to set up a subscription
     MTRWeakReference<MTRDevice *> * weakSelf = [MTRWeakReference weakReferenceWithObject:self];
-    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t) (kTimeToWaitBeforeMarkingUnreachableAfterSettingUpSubscription * NSEC_PER_SEC)), self.queue, ^{
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, static_cast<int64_t>(kSecondsToWaitBeforeMarkingUnreachableAfterSettingUpSubscription) * static_cast<int64_t>(NSEC_PER_SEC)), self.queue, ^{
         MTRDevice * strongSelf = weakSelf.strongObject;
         if (strongSelf != nil) {
-            os_unfair_lock_lock(&strongSelf->_lock);
-            [strongSelf _markDeviceAsUnreachableIfNotSusbcribed];
-            os_unfair_lock_unlock(&strongSelf->_lock);
+            std::lock_guard(strongSelf->_lock);
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/MTRDevice.mm:1065]:
```suggestion
            std::lock_guard lock(strongSelf->_lock);
```
I think this ought to fix it

Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTRDevice.mm:1065]: Oh, good catch!

File: src/darwin/Framework/CHIP/MTRDevice.mm (hunk 1 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTRDevice.mm:1066]: Compiled for me, but as @jtung-apple pointed out this is buggy.  :). Fixed now.

**Pull request merged**